app/ts/store/tokens.ts

```diff
@@ -0,0 +1,112 @@
+import { signal, useComputed, useSignalEffect } from '@preact/signals'
+import { useAccount } from './account.js'
+
+const CACHEID_PREFIX = '_ut'
+
+const retrieveTokensFromCache = (cacheKey: string) => {
+	const tokensCache = localStorage.getItem(cacheKey)
+
+	if (tokensCache === null) {
+		return DEFAULT_TOKENS
+	}
+
+	try {
+		return JSON.parse(tokensCache) as TokenMeta[]
```

Comment:
I think this should be properly parsed and not casted

Reply:
i now updated the function to validate parsed json from cache including assertions to narrow down the resulting array